### ISSUE TITLE
Setting debugShowCheckedModeBanner in the app is overwritten by the debugger

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#1474]
You can set `debugShowCheckedModeBanner` in code, but the debugger re-transmits its own values at startup.

**Comment by DanTup** [comment]
@pq How do you handle this in IntelliJ? Currently VS Code assumes a default value (of true) and then when you toggle it, it flips that and then sends it. However if the user has set it to false [in code](https://docs.flutter.io/flutter/material/MaterialApp/debugShowCheckedModeBanner.html) then the first toggle will be incorrect.

Do you detect the current value? I can't see anything obvious in your code that's doing that.

**Comment by pq** [comment]
IIRC we don't do anything smart here though I should confirm.

Do you have a simple test app I could use to take a look?

**Comment by DanTup** [comment]
@pq I'm just using the Gallery app but with debugShowCheckedModeBanner disabled in the MaterialApp:

<img width="719" alt="screenshot 2019-03-04 at 10 07 24 am" src="https://user-images.githubusercontent.com/1078012/53726173-5c241b80-3e65-11e9-9dfc-93ea85aa4d96.png">

Testing again, I think this issue is misleading - as in my testing the "toggle debug banner" in VS Code does nothing (the banner never shows). It was reported to me that something different was happening - that the app launched with the wrong setting (which I assumed was because we were re-sending it and stamping over the one in the code).

I guess the real question is how/whether these should play together. If the user disables it in code, what should happen? (I think ideally, we would *read* the value at startup, but then still let them toggle it - however I'm not sure if that's possible based on the testing I just did, since it never seems to re-appear).

**Comment by pq** [comment]
Ok, right.  Thanks for the added clarification.

> I guess the real question is how/whether these should play together. If the user disables it in code, what should happen? (I think ideally, we would read the value at startup, but then still let them toggle it - however I'm not sure if that's possible based on the testing I just did, since it never seems to re-appear).

We'd have to a little experimenting but it may not be a simple fix as you suspect.  For now perhaps best to document as a known limitation?

**Comment by DanTup** [comment]
Yeah, I'll move this to the backlog. I thought it'd be an easy fix, but I guess not. It's annoying because it's not obvious to a user why it's not working (and nobody reads docs). I'll update the title to make it more obvious this doesn't work, and maybe people will find this issue if they hit it.
